elastic-apm-agent/src/main/java/co/elastic/apm/agent/premain/ExcludeJvmBootstrapCheck.java

```diff
@@ -0,0 +1,174 @@
+ * <ul>
+ *     <li>
+ *         Example 1: allow JVM attachment only on Tomcat and some proprietary Java apps:
+ *         {@code -Delastic.apm.bootstrap_allowlist=org.apache.catalina.startup.Bootstrap*,my.cool.app.*}
+ *     </li>
+ *     <li>
+ *         Example 2: disable when some custom System properties are set:
+ *         {@code -Delastic.apm.bootstrap_exclude_list=custom.property.1,custom.property.2}
+ *     </li>
+ * </ul>
+ */
+public class ExcludeJvmBootstrapCheck implements BootstrapCheck {
+
+    private static final List<String> defaultExcludeList = Arrays.asList("activemq.home", "activemq.base");
+
+    @Nullable
+    private final String cmd;
+
+    @Nullable
+    private String allowListRaw;
+
+    @Nullable
+    private List<WildcardMatcher> configuredAllowList;
+
+    @Nullable
+    private String excludeListRaw;
+
+    @Nullable
+    private List<String> configuredExcludeList;
+
+    public ExcludeJvmBootstrapCheck(@Nullable String cmd) {
+        this.cmd = cmd;
+
+        allowListRaw = System.getProperty("elastic.apm.bootstrap_allowlist");
+        if (allowListRaw == null) {
+            allowListRaw = System.getenv("ELASTIC_APM_BOOTSTRAP_ALLOWLIST");
+        }
+        if (allowListRaw != null) {
+            configuredAllowList = parse(allowListRaw, new WildCardMatcherConverter());
+        }
+
+        excludeListRaw = System.getProperty("elastic.apm.bootstrap_exclude_list");
+        if (excludeListRaw == null) {
+            excludeListRaw = System.getenv("ELASTIC_APM_BOOTSTRAP_EXCLUDE_LIST");
+        }
+        if (excludeListRaw != null) {
+            configuredExcludeList = parse(excludeListRaw, new StringConverter());
+        }
+    }
+
+    @Override
+    public void doBootstrapCheck(final BootstrapCheckResult result) {
+        if (cmd != null && configuredAllowList != null) {
+            if (WildcardMatcher.isNoneMatch(configuredAllowList, cmd)) {
+                // when configuring allowlist, any JVM with non-matched command is implicitly excluded
+                result.addError(String.format("`elastic.apm.bootstrap_allowlist` or `ELASTIC_APM_BOOTSTRAP_ALLOWLIST` are configured with " +
+                "the pattern list '%s', which does not match this JVM's command: '%s'", allowListRaw, cmd));
+            }
+            // No need to keep looking if allow list configured
+            return;
+        }
+
+        final List<String> excludeSystemProperties = (configuredExcludeList != null) ? configuredExcludeList : defaultExcludeList;
+        if (System.getSecurityManager() == null) {
+            doExcludeListCheck(result, excludeSystemProperties);
+            return;
+        }
+
+        AccessController.doPrivileged(new PrivilegedAction<Void>() {
+            @Override
+            public Void run() {
+                doExcludeListCheck(result, excludeSystemProperties);
+                return null;
+            }
+        });
+
+        doExcludeListCheck(result, excludeSystemProperties);
```

Comment:
leftover ? we already execute it after checking if SM is present or not.

Reply:
Definitely looks like a leftover, thanks!